Remove dynamic config

dancormier opened the pull request:
This PR removes `lib/css/base/_stacks-configuration-dynamic.less`, and replaces all references to dynamic config Less variables to CSS variables (with the exception of `@body-site-width`, which has been replaced with the hard-coded value of `1060px`).

I've pulled these changes into Core, tried a few different themes, and have yet to notice any regressions.

dancormier committed:
Remove dynamic config

b-kelly commented on lib/css/base/_stacks-body.less:
```diff
@@ -33,10 +32,9 @@ body {
     box-sizing: border-box;
     min-height: 100%;
-    background-color: @body-background;
+    background-color: var(--white);
```
I'd recommend making this a `theme` variable. We use `--theme-background-color` in Core, so shadowing this would allow us to drop some legacy code there.

b-kelly commented on lib/css/base/_stacks-internals.less:
```diff
@@ -1,15 +1,5 @@
-    }
-
     #load-static-config() {
         #stacks-config-static();
```
We're coming for you next `#stacks-config-static`

b-kelly commented on lib/css/components/_stacks-banners.less:
```diff
@@ -73,9 +73,8 @@
     position: relative;
     width: 100%;
-    max-width: @body-site-width;
+    max-width: 1060px; // TODO consider creating a CSS variable for this
```
~~Definitely move this out to a variable. If we don't intend on allowing consumers to change this value, let's at least make it a Less variable so we can do math on it.~~

Wait, we're literally only using this variable in this one spot? Let's see if we can reuse a different variable instead. Perhaps we could use `@s-full`? That's what `.wmx12` and the topbar container uses by default.

dancormier replied:
I'm not sure why this was set to `1060px` to begin with. That value doesn't map to any existing value, so the banner will never be aligned with other content. I kept that value in place just to keep from changing **any_** values in this PR, but I think we'd be safe to tweak it.

How about `@s-step * 10`? That would be equivalent to `.wmx10` which is roughly `1054px`.

b-kelly replied:
I think mimicking `.wmx10` would be an excellent compromise. We can always revisit later if we want to.

b-kelly commented on lib/css/components/_stacks-buttons.less:
```diff
@@ -230,8 +230,7 @@
     .s-btn__muted {
-        #stacks-internals #load-config();
-        color: @button-muted-color;
+        color: var(--black-500);
```
Overall, lgtm. I'd like to consider making these theme variables, but I don't think that's necessary at the moment since we don't have any consumers overriding them.

b-kelly commented on lib/css/components/_stacks-tags.less:
```diff
@@ -166,46 +164,40 @@ a.s-tag:not(.is-selected) {
 .s-tag__required {
-    #stacks-internals #load-config();
-    .s-tag-styles(@tags-required-border, @tags-required-background, @tags-required-color);
+    .s-tag-styles(var(--bc-darker), var(--black-075), var(--black-700));
```
Same comment as above with buttons, lgtm.

b-kelly reviewed (changes requested):
Almost lgtm! Some small suggestions left inline.

b-kelly commented:
Side note: I'm _super duper stoked_ to finally see this happening!

dancormier committed:
Replace `@body-background` with `--theme` CSS var

Addresses https://github.com/StackExchange/Stacks/pull/882#discussion_r835340810

dancormier committed:
Replace custom value for banner container max width

b-kelly commented on lib/css/exports/_stacks-constants-colors.less:
```diff
@@ -198,7 +198,7 @@
 .component-colors {
     --theme-body-font-color: var(--black-800);
-
+    --theme-background-color: var(--white);
```
lgtm. This should "Just Work™" in Core, but I'll be sure to test for regressions there.

b-kelly reviewed (approved):
Code looks good to me. I haven't had a chance to really dive into a local/visual test yet, but I doubt there will be any issues there. I'll give that a go later and report back.
